kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStoreRegistration.java

```diff
@@ -1,42 +1,142 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package dev.responsive.kafka.store;
 
 import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
 import java.util.Objects;
+import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
+import java.util.function.LongSupplier;
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
 
 public final class ResponsiveStoreRegistration {
-  private final TopicPartition changelogTopicPartition;
-  private final long committedOffset;
+  public static final long NO_COMMITTED_OFFSET = -1L; // buffer is initialized but no prior offset
+  public static final long UNINITIALIZED_OFFSET = -2L; // buffer is not yet initialized
+
+  private final Logger log;
   private final String name;
+  private final TopicPartition changelogTopicPartition;
   private final Consumer<Long> onCommit;
 
+  private final LongSupplier storedOffset; // RPC to fetch offset committed in remote table
+  private final BooleanSupplier maybeInitialize; // returns true if/when initialization is done
+  private long startOffset; // stored offset as of initialization, ie where restore should start
+
+
+  @VisibleForTesting
```

Comment:
it looks like this was already here, but we should not use this annotation since it's pulled from a shaded guava - instead let's just add a comment

Reply:
Bruno was always the main proponent of this philosophy, but over time I've come to heartedly agree with him: that we should never add "//visible for testing" comments because they can't ever be trusted to be kept up to date.

My personal philosophy is, it's not worth making the distinction between public vs package-private for testing purposes, especially if it's not meant to be user-facing.

If you bumped something up from private to package-private  or public for testing purposes, then it's important to leave a comment -- but ideally you don't ever do that in the first place so it should be a non-issue

Will definitely remove the annotation though